CONTRIBUTING.md

```diff
@@ -53,6 +53,10 @@ In addition to writing properly formatted commit messages, it's important to inc
 
 For package version upgrades and such a one-line commit message is usually sufficient.
 
+We prefer not to use the "squash merge" feature in nixpkgs: in order to keep as much information as possible in the commit history, we expect pull requests to consist of self-contained commits as described above.
```

Comment:
What do you mean by ``to keep as much information as possible in the commit history``? The improved [squash merge by Refined GitHub](https://github.com/refined-github/refined-github/wiki/Extended-feature-descriptions#clear-pr-merge-commit-message) for example pastes all the commit titles into the squash merge body. The body would be lost but PRs that have well written commit messages are usually not squash merged.

Also we do not expect commits to be fully self contained. It is totally fine to also update meta.homepage, cleanup up an old ToDos or improve formatting in an update commit. Also nodePackages and vimPlugins do auto commits for many packages at once by design and are not really self-contained.